src/newmatrix.cpp

```diff
@@ -719,6 +719,29 @@ ACMD(do_matrix_position)
   WAIT_STATE(ch, (int) (DECKING_WAIT_STATE_TIME));
 }
 
+bool try_execute_shield_program(struct matrix_icon *icon, struct matrix_icon *targ, int &success)
+{
+  struct obj_data *soft = NULL;
+  for (soft = targ->decker->software; soft; soft = soft->next_content) {
+    if (GET_PROGRAM_TYPE(soft) == SOFT_SHIELD) {
+      int shield_test = success_test(GET_PROGRAM_RATING(soft), 
+        ICON_IS_IC(icon) ? matrix[icon->in_host].security : GET_SKILL(icon->decker->ch, SKILL_COMPUTER));
+
+      if (shield_test > 0) {
+        success -= shield_test;
+        send_to_icon(targ, "You raise your shield program and deflect some of the attack.\r\n");
+      }
+      GET_PROGRAM_RATING(soft)--;
+      if (GET_PROGRAM_RATING(soft) <= 0) {
+        send_to_icon(targ, "Your shield program crashes as the rating is depleted.\r\n");
+        extract_obj(soft);
```

Comment:
Does extracting it in the matrix like this permanently remove it, or does it just remove the loaded copy?

Also, is the rating decrease permanent, or does it only effect what's loaded at the time?

Reply:
It only removes the loaded copy, and it only affects the loaded copy.